src/main/java/domain/result/ClimbingResults.java

```diff
@@ -0,0 +1,57 @@
+package domain.result;
+
+import common.Command;
+import domain.ladder.Direction;
+import domain.ladder.Floor;
+import domain.ladder.Ladder;
+import domain.player.Player;
+import domain.player.Players;
+import domain.result.message.ResultExceptionMessage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClimbingResults {
+    private final Map<String, String> playerResults;
+
+    private ClimbingResults(final Map<String, String> playerResults) {
+        this.playerResults = Map.copyOf(playerResults);
+    }
+
+    public static ClimbingResults of(final Players players, final Ladder ladder, final LadderResults ladderResults) {
+        Map<String, String> results = new HashMap<>();
+        List<Floor> floors = ladder.getFloors();
+        for (Floor floor : floors) {
+            climbLadder(players, floor, ladderResults, results);
+        }
+        return new ClimbingResults(results);
+    }
+
+    private static void climbLadder(
+            final Players players,
+            final Floor floor,
+            final LadderResults ladderResults,
+            final Map<String, String> results) {
+        for (int i = 0; i < players.getPlayerCount(); i++) {
+            Player player = players.getPlayerOfIndex(i);
+            Direction bridgeLocation = floor.getBridgePosition(player.getPosition());
+            player.moveTo(bridgeLocation.getValue());
+            results.put(player.getName(), ladderResults.getLadderResultOfIndex(player.getPosition()));
+        }
```

Comment:
"불필요한 객체 생성을 피하라"라는 내용에서 똑같은 기능을 가진 객체를 생성하기 보다는 객체 하나를 재사용하라고 이야기하고 있어
반복문을 돌 때마다 똑같은 기능을 가진 Map이 생성되는 것을 막고 싶어서 인자로 넘기는 방법을 생각했었습니다.
제가 아직 이러한 내용에 대해 깊이 있는 이해가 어려워서 지금 상황에서는 어떤 방법이 좋은 방법인지 많이 헷갈리네요..
지금같이 indent가 1인 제한 상황에서는 파라미터인자를 여러개 넘겨 코드를 이해하기 어렵게 하기 보다는 줄일 수 있는 방법으로 작성하는 것이 최선일까요?

Reply:
"불필요한 객체 생성을 피하라"는 가독성을 버리고서라도 지킬 필요는 없습니다.
객체가 여러 번 생성되는 코드지만 가독성이 좋아진다면 그건 "불필요한" 생성이 아니겠죠.

조금 다른 맥락(불변 객체)에서 나온 얘기긴 하지만 오라클에서는 이렇게 말합니다 [링크](https://docs.oracle.com/javase/tutorial/essential/concurrency/immutable.html)

> Programmers are often reluctant to employ immutable objects, because they worry about the cost of creating a new object as opposed to updating an object in place. The impact of object creation is often overestimated, and can be offset by some of the efficiencies associated with immutable objects. These include decreased overhead due to garbage collection ...

코드를 구현할 때 너무 객체 생성의 효율성을 따지지 않아도 괜찮습니다.

Reply:
불 필요한 객체 생성 때문에 불변 객체를 꺼리는 프로그래머가 있지만
가비지 컬렉션으로 인한 오버헤드가 감소하고, 객체의 상태를 보호하기 위한 추가적인 코드를 작성하지 않아도 된다는 이점이 있어
불변 객체를 사용했을 때 불 필요한 객체 생성을 상쇄하는 이점이 있다는 내용이네요!

상황에 따라 유연할 줄알아야하는데 "불필요한 객체 생성을 피하라"라는 문장에 너무 매몰되었던 것 같습니다
범블비덕에 좋은 인사이트 얻어갑니다!